mlflow/pytorch/_lightning_autolog.py

```diff
@@ -20,9 +20,28 @@
 logging.basicConfig(level=logging.ERROR)
 MIN_REQ_VERSION = Version(_ML_PACKAGE_VERSIONS["pytorch-lightning"]["autologging"]["minimum"])
 MAX_REQ_VERSION = Version(_ML_PACKAGE_VERSIONS["pytorch-lightning"]["autologging"]["maximum"])
-
-import pytorch_lightning as pl
-from pytorch_lightning.utilities import rank_zero_only
+# Flag for when we're using legacy 'pytorch-lightning' import name
+# instead of 'lightning'; this can be deprecated once 'pytorch-lightning' is no longer supported.
+PYTORCH_LIGHTNING_LEGACY_NAMESPACE = False
+
+# Handle import of either 'pytorch_lightning' or 'lightning'
+try:
+    import lightning.pytorch as pl
+    from lightning.pytorch.utilities import rank_zero_only
+except ModuleNotFoundError:
+    logging.warning("'lightning' package not found, attempting to use 'pytorch_lightning instead.'")
+    # TODO : Using a nested try/catch isn't always nice, might be a better way to handle this?
+    try:
+        import pytorch_lightning as pl
+        from pytorch_lightning.utilities import rank_zero_only
+
+        PYTORCH_LIGHTNING_LEGACY_NAMESPACE = True
```

Comment:
Add a warning here instead
_logger.warning("'pytorch_lightning' will be deprecated in a future release, please use 'lightning' instead. `pip install lightning`)